routing/payment_session.go

```diff
@@ -135,10 +129,10 @@ func (p *paymentSession) RequestRoute(payment *LightningPayment,
 	route, err := newRoute(
 		sourceVertex, path, height,
 		finalHopParams{
-			amt:         payment.Amount,
+			amt:         maxAmt,
```

Comment:
When do you want to address setting the final hop amt to `payment.Amount`? `newRoute` really needs to take an extra amount separate from the final hop amount.

Reply:
Leaving for #3967

Reply:
Yes, implemented there